src/main/java/com/epam/ta/reportportal/ws/rabbit/LaunchReporterConsumer.java

```diff
@@ -52,36 +57,92 @@
 
 	private FinishLaunchHandler finishLaunchHandler;
 
+	private AmqpTemplate amqpTemplate;
+
 	@Autowired
 	public LaunchReporterConsumer(DatabaseUserDetailsService userDetailsService, StartLaunchHandler startLaunchHandler,
-			FinishLaunchHandler finishLaunchHandler) {
+			FinishLaunchHandler finishLaunchHandler, @Qualifier("rabbitTemplate") AmqpTemplate amqpTemplate) {
 		this.userDetailsService = userDetailsService;
 		this.startLaunchHandler = startLaunchHandler;
 		this.finishLaunchHandler = finishLaunchHandler;
+		this.amqpTemplate = amqpTemplate;
 	}
 
 	@RabbitListener(queues = "#{ @launchStartQueue.name }")
 	public void onStartLaunch(@Payload StartLaunchRQ rq, @Header(MessageHeaders.USERNAME) String username,
-			@Header(MessageHeaders.PROJECT_NAME) String projectName) {
-		ReportPortalUser userDetails = (ReportPortalUser) userDetailsService.loadUserByUsername(username);
-		startLaunchHandler.startLaunch(userDetails, ProjectExtractor.extractProjectDetails(userDetails, projectName), rq);
+			@Header(MessageHeaders.PROJECT_NAME) String projectName,
+			@Header(required = false, name = MessageHeaders.XD_HEADER) List<Map<String, ?>> xdHeader) {
+		if (xdHeader != null) {
+			long count = (Long) xdHeader.get(0).get("count");
+			if (count > DEAD_LETTER_MAX_RETRY) {
+				LOGGER.error("Dropping to {} start request for Launch {}, on maximum retry attempts {}",
+						QUEUE_LAUNCH_START_DLQ_DROPPED,
+						rq.getUuid(),
+						DEAD_LETTER_MAX_RETRY);
+
+				amqpTemplate.convertAndSend(QUEUE_LAUNCH_START_DLQ_DROPPED, rq, message -> {
+					Map<String, Object> headers = message.getMessageProperties().getHeaders();
+					headers.put(MessageHeaders.USERNAME, username);
+					headers.put(MessageHeaders.PROJECT_NAME, projectName);
+					return message;
+				});
+
+				return;
+			}
+			LOGGER.trace("Retrying start request for Launch {}, attempt {}", rq.getUuid(), count);
+		}
+		try {
+			ReportPortalUser userDetails = (ReportPortalUser) userDetailsService.loadUserByUsername(username);
+			startLaunchHandler.startLaunch(userDetails, ProjectExtractor.extractProjectDetails(userDetails, projectName), rq);
+		} catch (Exception e) {
```

Comment:
let's configure exception logging on some spring (if possible) or aop layer

Reply:
it's high time to do, added )